pandas/core/window/rolling.py

```diff
@@ -544,19 +533,32 @@ def aggregate(self, func, *args, **kwargs):
     agg = aggregate
 
 
-class BaseWindowGroupby(GotItemMixin, BaseWindow):
+class BaseWindowGroupby(BaseWindow):
     """
     Provide the groupby windowing facilities.
     """
 
-    def __init__(self, obj, *args, **kwargs):
-        kwargs.pop("parent", None)
-        groupby = kwargs.pop("groupby", None)
-        if groupby is None:
-            groupby, obj = obj, obj._selected_obj
-        self._groupby = groupby
-        self._groupby.mutated = True
-        self._groupby.grouper.mutated = True
+    _attributes = [
```

Comment:
rather than doing this. can you create a private classmethod, instantiate and mutate the returned object? e.g.

```
@classmethod _create_from_grouping(cls,, grouping_indices, keys, codes......):
    obj = cls(.......)
    obj._grouping_indices = grouping_indcies
    .....
    return obj
```

Reply:
This attribute (`_attributes`) is currently necessary to the window classes for `_gotitem` where a shallow copy is performed, so I would still need to keep this reference

https://github.com/pandas-dev/pandas/pull/39765/files#diff-376f031d24a9c167d8ea06982971b613702f7dd122e168c3a8b333009a811ed4R254